### PR TITLE
feat: implement database indexing and premium UI for historical game …

### LINKED ISSUE #554: Backend: Database indexing for historical game search
## 📖 Context
Enhance the user experience with premium Next.js UI components and fluid Web3 interactions.

## 🛠️ Tasks
- Analyze the existing code in `frontend/app/`.
- Implement the feature following our established design patterns.
- Ensure efficient resource utilization (Gas/CPU).

## ✅ Acceptance Criteria
- Code is well-documented and follows style guides.
- Unit tests cover standard and edge cases.
- Feature is fully integrated and tested with `cargo` or `npm`.

### PR DESCRIPTION
**Pull request by AbuJulaybeeb**
Implemented advanced historical game search functionality across the stack. This includes database-level optimizations for high-speed queries and a premium Next.js UI for the user dashboard.
## Changes
- **Database**: Added migrations for `result`, `variant`, and composite head-to-head indexes in the `smdb` schema.
- **API/Service**: Enhanced the `list_games` endpoint with advanced filtering parameters (opponent, result, variant, date range).
- **UI/UX**: Updated the `RecentMatches` dashboard component with a high-contrast search and filter system, ensuring a premium feel consistent with XLMate's aesthetic.
## Verification
- **Backend**: Verified query logic via `MockDatabase` unit tests in `GameService`.
- **Frontend**: Verified UI responsiveness and filtering logic in the Dashboard.
closes #554 

## Summary by CodeRabbit

* **New Features**
  * Advanced game-history filters: search by opponent, variant, result (win/loss/draw), and date range.
  * Interactive match history UI: opponent search, outcome filter, updated headers, empty-state and reset controls.

* **Performance**
  * Added indexes to speed historical and head‑to‑head game searches and variant/result lookups.

### CODE REVIEW

File: backend/modules/service/src/games.rs [hunk 1]
```diff
@@ -170,22 +170,76 @@ impl GameService {
...
+            // Map result string to enum
+            let r_enum = match r.as_str() {
+                "white_wins" => db_entity::game::ResultSide::WhiteWins,
+                "black_wins" => db_entity::game::ResultSide::BlackWins,
+                "draw" => db_entity::game::ResultSide::Draw,
+                "abandoned" => db_entity::game::ResultSide::Abandoned,
+                _ => db_entity::game::ResultSide::Ongoing,
+            };
+            query = query.filter(game::Column::Result.eq(r_enum));
+        }
```
[the remark this answers is not in the document]
Comment (AbuJulaybeeb) [reply backend/modules/service/src/games.rs:247]: fixed
